### PR TITLE
Forward port #7428

### PR DESCRIPTION
**Pull request by rchavik**
Need to pull @markstory's test commit, but I'll leave this here for now.

### CODE REVIEW

File: src/Database/Driver.php [hunk 1]
```diff
@@ -283,7 +283,17 @@ public function lastInsertId($table = null, $column = null)
      */
     public function isConnected()
     {
-        return $this->_connection !== null;
+        if ($this->_connection === null) {
+            $connected = false;
+        } else {
+            try {
+                $connected = $this->_connection->query('SELECT 1');
+            } catch (\PDOException $e) {
```
Comment (lorenzo) [inline src/Database/Driver.php]: Only PDO drivers should care about PDO

Comment (rchavik) [inline src/Database/Driver.php]: Ah, so move this to PDODriverTrait() ?

Comment (lorenzo) [inline src/Database/Driver.php]: Yeah

**Comment by markstory** [comment]
@rchavik ping.

**Comment by rchavik** [comment]
i'll fix this up this weekend :)

**Comment by rchavik** [comment]
I've moved the `isConnected()` implementation to PDODriverTrait and cherry-picked markstory's c14d1ffe9390cb8f2f67146fe1045fff254a2340. It does not exactly test for my use case, ie: long running worker scripts that lost connection to mysql. Not sure how to simulate lost connection.

**Comment by lorenzo** [comment]
Thanks @rchavik :)

**Pull request merged**